Client/Frontend/Browser/Playlist/PlaylistInfo.swift

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import WebKit
+import Shared
+
+private let log = Logger.browserLogger
+
+struct PlaylistInfo: Codable {
+    let name: String
+    let src: String
+    let pageSrc: String
+    let pageTitle: String
+    let mimeType: String
+    let duration: Float
+    let detected: Bool
+    let dateAdded: Date
+    
+    init(item: PlaylistItem) {
+        self.name = item.name ?? ""
+        self.src = item.mediaSrc ?? ""
+        self.pageSrc = item.pageSrc ?? ""
+        self.pageTitle = item.pageTitle ?? ""
+        self.mimeType = item.mimeType ?? ""
+        self.duration = item.duration
+        self.dateAdded = item.dateAdded ?? Date()
```

Comment:
Do you prefer to add default values to these properties instead of making them optional?
I guess it makes sense since we have to present something in UI

Reply:
CoreData makes all fields nullable by default even if they are specified as non-null. It still generates the file with all optional fields. So I have to unwrap it anyway.